### ISSUE TITLE
Array of counts doesn't work in multinomial mixture

### ISSUE THREAD

**Issue by 35ajstern** [bayespy/bayespy#122]
Hi there,

I'm trying to set up a binomial mixture model where the dataset `x` is NxD, taking on positive integer values. (There is a second dataset of the same dimensionality, `counts`, which is the number of draws for the corresponding element of `x`. I.e., 0 ≤ x ≤ counts.) I want to model each element in X as a binomial random variable who probability is specified by a mixture of the latent groups labeled 1...K. Here is my starter code, in which I use N=100, K=2, and D=1000: 

```
import numpy as np
import bayespy as bp

#### data generation & params 

# num samples
N = 100
# true number of groups
K = 2
# number of columns in dataset
D = 1000
# beta distn params
a = 1
b = 1
# randomly draw group proportions
pi = np.random.dirichlet(0.5*np.ones(K))
assert(len(pi)==K)
# groupwise column probabilities
p = np.random.beta(a,b,size=(K,D))

from bayespy.utils import random

rate = 20.0
count = 1+np.random.poisson(rate,size=(N,D))
z = random.multinomial(count, pi)
z = np.rollaxis(z,2)
props = z/z.sum(axis=0)
props = props.T
props = np.swapaxes(props,0,1)

# draw number of alt reads per individual per site
x = random.multinomial(count,props)

##### inference with bayespy

from bayespy.nodes import Categorical, Dirichlet, Multinomial, Beta
R = Dirichlet(1e-3*np.ones(K),
              name='R')

Z = Multinomial(count,
                R,
                name='Z')

from bayespy.nodes import Beta
P = Beta([a, b],
    plates=(D,Knom),
    name='P')

from bayespy.nodes import Mixture, Bernoulli, Binomial

X = Mixture(Z, Multinomial, count, P)
```

when I set up the mixture model in the final line, I get the following error: `ValueError: The plates of the parents do not broadcast: ((1000, 10), (100, 1000))`

Any idea how to correct this?

Thanks!

**Comment by jluttine** [comment]
Hi, sorry for the delay. I looked into this. There were a few small errors in your code. I did small modifications, you can `diff` the two code snippets to see what I changed.

```
import numpy as np
import bayespy as bp

#### data generation & params 

# num samples
N = 100
# true number of groups
K = 2
# number of groups in the model
Kest = 4
# number of columns in dataset
D = 1000
# beta distn params
a = 1
b = 1
# randomly draw group proportions
pi = np.random.dirichlet(0.5*np.ones(K))
assert(len(pi)==K)
# groupwise column probabilities
p = np.random.beta(a,b,size=(K,D))

from bayespy.utils import random

rate = 20.0
count = 1+np.random.poisson(rate,size=(N,D))
z = random.multinomial(count, pi)
z = np.rollaxis(z,2)
props = z/z.sum(axis=0)
props = props.T
props = np.swapaxes(props,0,1)

# draw number of alt reads per individual per site
x = random.multinomial(count,props)

##### inference with bayespy

from bayespy.nodes import Categorical, Dirichlet, Multinomial, Beta
R = Dirichlet(1e-3*np.ones(Kest),
              name='R')

Z = Categorical(R,
                plates=(N, D),
                name='Z')

from bayespy.nodes import Beta
P = Beta([a, b],
    plates=(Kest,),
    name='P')

from bayespy.nodes import Mixture, Bernoulli, Binomial

X = Mixture(Z, Multinomial, count[...,None], P)
```

This is how I would've hoped it to work, but it turns out it doesn't. There seems to be an issue when using an array as `count`s in Multinomial mixture, so it's a bug in BayesPy. I couldn't come up with a simple quick fix for it unfortunately, I'll need to think more. 

As a note to myself: The issue is that the mixture node sums over the mixed plate axis, but as `count` isn't treated as a normal parameter in multinomial node, it doesn't get this treatment and therefore the shape doesn't match in computations that are performed after the mixed plate axis has been summed over..

**Comment by jluttine** [comment]
This is now fixed in 0.5.22.

Here's a quick sketch on how to make it work:

```python
import numpy as np
import bayespy as bp

#### data generation & params 

# num samples
N = 10000
# true number of groups
K = 3
# number of columns in dataset
D = 10
# beta distn params
a = 1
b = 1

# randomly draw group proportions
pi = np.random.dirichlet(1.0 * np.ones(K))

# Assignments to groups
z = np.random.randint(0, K, size=(N, D))

# Groupwise column probabilities.
# Each column has different probabilities for the binomial.
p = np.random.beta(a, b, size=(D, K))

rate = 20.0
# Number of trials in the binomial for each element
count = 1 + np.random.poisson(rate, size=(N, D))

# Pick the binomial probability for each element
# based on the group assignments
props = np.take_along_axis(p[None,:,:], z[:,:,None], axis=-1)
assert props.shape == (N, D, 1)
props = np.squeeze(props, axis=-1)

# Draw binomial elements
x = np.random.binomial(count, props, size=(N, D))

##### inference with bayespy

from bayespy.nodes import Categorical, Dirichlet, Multinomial, Beta
R = Dirichlet(1e-3*np.ones(K),
              name='R')

Z = Categorical(R,
                plates=(N, D),
                name='Z')
Z.initialize_from_random()

from bayespy.nodes import Beta
P = Beta([a, b],
    plates=(D, K),
    name='P')

from bayespy.nodes import Mixture, Bernoulli, Binomial

X = Mixture(Z, Binomial, count[...,None], P)

X.observe(x)

from bayespy.inference import VB

Q = VB(X, P, Z)
Q.update(repeat=100)
```

Not sure if it's exactly what you wanted, but it should serve as a starting point at least.
